app/lib/redis.server.ts

```diff
@@ -0,0 +1,28 @@
+import Redis from 'ioredis';
+
+import type { Redis as RedisType, RedisOptions } from 'ioredis';
+
+let redis: RedisType;
+
+declare global {
+  var __redis: RedisType | undefined;
```

Comment:
Why do we do this global, if we also later export redis in line 5?

Reply:
NOTE: this code is coming from https://github.com/remix-run/examples/tree/main/bullmq-task-queue.

I think they do it this way so that they only put it on the global if not in production.  So it's module-level unless it needs to get elevated to the global.

Reply:
Okay, so thos only affects hot reload, Thanks!